client/components/global-notices/index.jsx

```diff
@@ -86,21 +89,25 @@ const NoticesList = createReactClass( {
 		//and from the old component. When all notices are moved to redux store, this component
 		//needs to be updated.
 		noticesList = noticesList.concat(
-			this.props.storeNotices.map( function( { button, href, onClick, ...notice } ) {
-				return (
-					<Notice
-						{ ...notice }
-						key={ `notice-${ notice.noticeId }` }
-						onDismissClick={ this.removeReduxNotice( notice ) }
-					>
-						{ button && (
-							<NoticeAction href={ href } onClick={ onClick }>
-								{ button }
-							</NoticeAction>
-						) }
-					</Notice>
-				);
-			}, this )
+			this.props.storeNotices.map(
+				// Collect `<Notice />` props in rest, extract other expected props
+				function( { button, href, noticeId, onClick, ...notice } ) {
+					return (
+						<Notice
+							{ ...notice }
+							key={ `notice-${ noticeId }` }
+							onDismissClick={ this.removeReduxNotice( noticeId, notice.onDismissClick ) }
```

Comment:
Nit question: Should the `onDismissClick` be extracted as well?

Reply:
Makes sense, I think. The less we rely on the order of props (our own `onDismissClick` overriding the one currently passed as part of `{ ...notice }`), the better.